sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -170,13 +170,52 @@ bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
       "OpenCL API failed. OpenCL API returns: " + codeToString(Error), Error);
 }
 
+bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
+                                const NDRDescT &NDRDesc) {
+  const bool HasLocalSize = (NDRDesc.LocalSize[0] != 0);
+
+  const plugin &Plugin = DeviceImpl.getPlugin();
+  RT::PiDevice Device = DeviceImpl.getHandleRef();
+
+  if (HasLocalSize) {
+    size_t maxThreadsPerBlock[3] = {};
+    Plugin.call<PiApiKind::piDeviceGetInfo>(
+        Device, PI_DEVICE_INFO_MAX_WORK_ITEM_SIZES, sizeof(maxThreadsPerBlock),
+        maxThreadsPerBlock, nullptr);
+
+    for (size_t i = 0; i < 3; ++i) {
+      if (maxThreadsPerBlock[i] < NDRDesc.LocalSize[i]) {
```

Comment:
Why this check cannot be applied for OpenCL as well?

Reply:
Good point. What we could do is to do general checks first, then at the end (if this check didn't fail) we either report unknown error or go through backend specific handling (i.e. `oclHandleInvalidWorkGroupSize` for OpenCL and currently nothing for CUDA).

Reply:
Sounds good.

Reply:
I have made this change. Thank you!